tests/test_yield.py

```diff
@@ -0,0 +1,14 @@
+
+def test_error_during_setup(pytester):
+    pytester.copy_example("examples/test_yield.py")
+    result = pytester.runpytest('-v', '-s')
+    result.assert_outcomes(passed=3)
+    result.stdout.re_match_lines(
+        [
+            ".*test_yield.a.*",
+            "setup",
+            ".*test_yield.b.*",
+            ".*test_yield.c.*",
+            "teardown",
+        ]
+    )
```

Comment:
An extra test maybe like this (I'm not so sure of the in's and outs of tests) would ensure `test_yield_extra_teardown` is failing,.

```suggestion
    )

def test_error_during_teardown(pytester):
    """
    Teardown exception should cause
    - all tests to pass
    - last test to error
    - yes, this is normal-ish for pytest with parametrized errors.
    """
    pytester.copy_example("examples/test_yield.py")
    result = pytester.runpytest('test_yield.py::test_yield_extra_teardown', '-v', '-s')
    result.assert_outcomes(passed=3, errors=1)
```

Reply:
agree that this is needed if we allow a teardown function. But I think I'll remove the option